packages/rum-core/src/domain/configuration.ts

```diff
@@ -55,8 +59,13 @@ export function validateAndBuildRumConfiguration(
     return
   }
 
-  if (initConfiguration.replaySampleRate !== undefined && !isPercentage(initConfiguration.replaySampleRate)) {
-    display.error('Replay Sample Rate should be a number between 0 and 100')
+  // TODO remove me in next major
+  if (initConfiguration.replaySampleRate !== undefined && initConfiguration.premiumSampleRate === undefined) {
+    initConfiguration.premiumSampleRate = initConfiguration.replaySampleRate
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏we usually don't mutate the InitConfiguration, to have a more declarative approach. Maybe use something like:
```ts
premiumSampleRate: initConfiguration.premiumSampleRate ?? initConfiguration.replaySampleRate ?? 100,
```